docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +165,162 @@ Note that:
+  ]
+```
+
+## Details to Aspect 2: Filetransfer from/to external device
+
+For aspect (2) there is a proposal, followed by another alternative proposal. Decision has to been taken which proposal to follow.
+
+### Proposal: Filetransfer provided by tedge_agent, based on HTTP
+
+To provide/consume configuration files to/from external devices, the HTTP filetransfer feature of the `tedge_agent` is used.
+  
+The HTTP filetransfer feature of the `tedge_agent` provides the service to transfer files from external devices to the local filesystem of the thin-edge device's, and vice versa.
+
+  * `tedge_agent` serves PUT and GET requests on `http://<thin-edge IP address>/tedge/filetransfer/<path>`
+    * `path` is chosen by the external-device; it may contain folders (e.g. `foo/bar/config-file.conf`)
+    * on an incoming PUT request `tedge_agent` stores the contained file to `/http-root/<path>`
+    * on an incoming GET request `tedge_agent` sends the file `/http-root/<path>` to the requester (i.E. the external device)
+    * `tedge_agent` does not force any predefined folder structure within `/http-root/`. However for configuration management it is best practice to use folder structure as below:
+      * `/http-root/config/current/<child-id>/<config filename>`<br/>
+         for config-files provided by the external device to the thin-edge device (current state - or latest known state)<br/>
+      * `/http-root/config/desired/<child-id>/<config filename>`<br/>
+        for config-files to be consumed by the external device (desired state)
+  * `tedge_agent` notifies over MQTT any change of a file under `/http-root/` (e.g. when uploaded by a external-device or updated by a local process). The notification messages are published on the topic `tedge/filetransfer_change/{path}`.<br/>
+    TO-BE-DEFINED: Content of payload?
+      
+      Example:
+      * HTTP PUT URL: `http://<thin-edge IP address>/tedge/filetransfer/config/current/child1/bar.conf`
+      * relating MQTT notification topic: `/tedge/filetransfer_change/config/current/child1/bar.conf`
+
+  * `c8y_configuration_plugin` allows to handle two versions for each configuration file (_current_ and _desired_).
+    * The configuration file's record `path` is treated as path of the _current_ version of a configuration file (see to section 'Configuration' above).
+    * The configuration file's record `desired` can be used to define the path of the _desired_ version of a configuration file. If `desired` is not defined, then `path` is treated as _desired_ version also.
+    * On c8y read request for a configuration, `c8y_configuration_plugin` returns the `current` file content.
+    * On c8y write request for a configuration, `c8y_configuration_plugin` updates the `desired` file content and waits for the `current` content to be updated to return the operation status to cumulocity.  
```

Comment:
I could not understand why we need `current` and `desired` directories. Can you write two sequence diagram to describe how tedge device and external device communicates, one for "config upload request" and the other for "config download request"? It's quite difficult to understand the flow.

Reply:
That `current` and `disired` approach was meant to check after a configuration update from the cloud, whether the new configuration was accpepted by the external device. 

However that approach was now removed from the spec, as we switched to a concept where the config-file from all external devices are no more kept on the thin-edge device, but requested on purpose.

We introduced now a notification for the configuration update, where the external device reports to the thin-edge device whether the new configuration was accepted or not.